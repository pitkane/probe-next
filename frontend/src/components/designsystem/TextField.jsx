import React from 'react';

const TextField = (props) => {
  const { required, name, placeholder, type, icon, onChange } = props;

  return (
    <div className="uk-margin">
      <div className="uk-inline uk-width-1-1">
        <span className="uk-form-icon uk-form-icon-flip" data-uk-icon={icon ? `icon: ${icon}` : ''} />
        <input onChange={onChange} name={name} className="uk-input uk-form-large" required={required} placeholder={placeholder} type={type} />
      </div>
    </div>
  );
};

TextField.defaultProps = {
  required: false,
};

export default TextField;
