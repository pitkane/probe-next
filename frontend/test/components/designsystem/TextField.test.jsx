import React from 'react';
import renderer from 'react-test-renderer';

import TextField from '../../../src/components/designsystem/TextField';

describe('TextField', () => {
  it('renders correctly', () => {
    const component = (
      <TextField />
    );
    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
