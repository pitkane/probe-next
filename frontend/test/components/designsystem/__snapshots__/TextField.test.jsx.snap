// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField renders correctly 1`] = `
<div
  className="uk-margin"
>
  <div
    className="uk-inline uk-width-1-1"
  >
    <span
      className="uk-form-icon uk-form-icon-flip"
      data-uk-icon=""
    />
    <input
      className="uk-input uk-form-large"
      required={false}
    />
  </div>
</div>
`;
